core-bundle/src/Resources/contao/library/Contao/InsertTags.php

```diff
@@ -436,6 +436,23 @@ protected function doReplace($strBuffer, $blnCache)
 							break;
 						}
 
+						$strName = $objNextPage->title;
+						$strTarget = $objNextPage->target ? ' target="_blank" rel="noreferrer noopener"' : '';
+						$strClass = $objNextPage->cssClass ? sprintf(' class="%s"', $objNextPage->cssClass) : '';
+						$strTitle = $objNextPage->pageTitle ?: $objNextPage->title;
+
+						// Early return for tags that do not need the URL
+						switch (strtolower($elements[0]))
+						{
+							case 'link_title':
+								$arrCache[$strTag] = StringUtil::specialcharsAttribute($strTitle);
+								break 2;
+
+							case 'link_name':
+								$arrCache[$strTag] = StringUtil::specialcharsAttribute($strName);
+								break 2;
+						}
+
```

Comment:
I think we could get rid of the duplicated code (and the `break 2`s) if we instead use something like this:

```php
						// Do not generate URLs for tags that do not need it
						if (in_array(strtolower($elements[0]), array('link_title', 'link_name')))
						{
							$strUrl = null;
						}
						else
						{
							// The existing switch ($objNextPage->type) block
						}

```

Reply:
I think early return is better and easier to read than yet another level of [cyclomatic_complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity). The method/case is already long enough and hard to understand whats happening where …

Reply:
IMO `break 2` and the duplicated code is way harder to unserstand.